### ISSUE TITLE
urllib3 v1.25 support

### ISSUE THREAD

**Issue by EpicWink** [kennethreitz/requests#5067]
Support for `urllib3` [v1.25](https://github.com/urllib3/urllib3/releases/tag/1.25) (released today/yesterday). The usual yada-yada "my CI is broken".

Multiple packages that we depend on (eg `google.cloud.storage`, `azure.storage`) have strict version-check on `requests` using `pkg_resources.require('requests >= 2.18.0')`. Any unmet dependencies in the chain below and including `requests` will cause the check to fail and raise an exception. `urllib3<1.25` is currently one such dependency.

## Expected Result
No exception raised on import

## Actual Result
```
pkg_resources.ContextualVersionConflict: (urllib3 1.25 (/usr/local/lib/python3.6/dist-packages), Requirement.parse('urllib3<1.25,>=1.21.1'), {'requests'})

The above exception was the direct cause of the following exception:
ImportError: ``requests >= 2.18.0`` is required by the ``google.resumable_media.requests`` subpackage.
It can be installed via
    pip install google-resumable-media[requests].
```

## Reproduction Steps
```bash
pip3 install sentry-sdk google-cloud-storage
```
```python
>>> from google.cloud import storage
```

## System Information
```
/home/laurie/env-tmp/lib/python3.6/site-packages/requests/__init__.py:91: RequestsDependencyWarning: urllib3 (1.25) or chardet (3.0.4) doesn't match a supported version!
  RequestsDependencyWarning)
{
  "chardet": {
    "version": "3.0.4"
  },
  "cryptography": {
    "version": ""
  },
  "idna": {
    "version": "2.8"
  },
  "implementation": {
    "name": "CPython",
    "version": "3.6.7"
  },
  "platform": {
    "release": "4.18.0-17-generic",
    "system": "Linux"
  },
  "pyOpenSSL": {
    "openssl_version": "",
    "version": null
  },
  "requests": {
    "version": "2.21.0"
  },
  "system_ssl": {
    "version": "1010100f"
  },
  "urllib3": {
    "version": "1.25"
  },
  "using_pyopenssl": false
}
```

**Comment by EpicWink** [comment]
Note that a pull-request fixing this should close #5065 and #4961

**Comment by EpicWink** [comment]
Also note that a solution to the above reproduction example is:
```bash
pip3 install sentry-sdk google-cloud-storage 'urllib3<1.25'
```

**Comment by Freso** [comment]
There’s already a PR in the works: https://github.com/kennethreitz/requests/pull/5063

**Comment by EpicWink** [comment]
To those watching, as [mentioned](https://github.com/kennethreitz/requests/pull/5063#issuecomment-487436504), we're [blocked](https://github.com/kennethreitz/requests/issues/5065#issuecomment-487423564) on `urllib3` v1.25.2

**Comment by sethmlarson** [comment]
@nateprewitt FYI 1.25.2 will be released very soon, I would recommend in addition to disallowing 1.25.0 also disallow 1.25.1?

**Comment by rkooo567** [comment]
Our CI pipeline starts to get an error `[integration_py3_docker_metric] 19-04-29:10:26:57 ERROR [clipper_metric_docker.py:127] Failed to parse: http://localhost:44328/api/v1/series?match[]=clipper_mc_parse_time_ms_sum` from April 22nd, from when `urllib3` version 1.25 was released. Do you think this is related to this issue?

**Comment by EpicWink** [comment]
@rkooo567, have you tried running the CI with the additional requirement `urllib3<1.25`?

**Comment by rkooo567** [comment]
@EpicWink Yes. And it seems like it resolves an issue. I also found out `urllib3==1.25.2` also resolves the problem.

**Comment by sethmlarson** [comment]
urllib3 1.24.3 has been released which fixes only the CRLF injection issue for 1.24.X users.

**Comment by plowman** [comment]
Here's a possibly dumb question: why does requests restrict urllib3 to be < 1.25? Since many other pip dependencies are going to continue to increase their minimum allowed urllib3 version, it seems like this is going to guarantee two things:
1. Every minor version bump in urllib3 will break many projects which depend on requests (e.g. anyone who is using the major Google or Microsoft Azure packages, among many others).
2. Every minor version bump in urllib3 will then require a new release of requests to allow the new version of urllib3.

Does urllib3 make breaking changes that often that we're worried about just trusting their versioning? Or is there another concern?

**Comment by yjqiang** [comment]
@plowman https://github.com/kennethreitz/requests/issues/5080 Looks like [the update to version 1.25 would make requests start up much slower](https://github.com/urllib3/urllib3/issues/1590#issue-438508993).

**Comment by Freso** [comment]
Based on the number of references to this issue in other projects’ trackers, I’m sure there are a lot of other project maintainers that would appreciate a resolution to this sooner rather than later… :) (Pointing it out since issue references don’t trigger e‐mail notifications.)

**Comment by tucked** [comment]
+1 because it aggravates problems caused by [Pip's lack of a dependency resolver](https://github.com/pypa/pip/issues/988) in some cases: https://github.com/pradyunsg/zazo/issues/14

**Comment by prometheanfire** [comment]
If you think that's fun, look at all the things that use requests and can't update urllib3.  (we do some manual checking)

http://logs.openstack.org/36/658636/3/check/requirements-tox-py27-check-uc/48d2334/job-output.txt.gz#_2019-05-14_06_33_55_887702

**Comment by nateprewitt** [comment]
Resolved with #5092, release is up on pypi now. Thanks again @EpicWink
